src/status_im/ui/screens/accounts/events.cljs

```diff
@@ -157,6 +158,14 @@
       {:db            (assoc-in db [:accounts/accounts id] new-account)
        ::save-account new-account})))
 
+(handlers/register-handler-fx
+  :account-update-wallet-settings
+  (fn [{{:accounts/keys [current-account-id accounts] :as db} :db} [_ settings]]
+    (let [current-account (get accounts current-account-id)
+          new-account (assoc current-account :settings settings)]
+      {:db            (assoc-in db [:accounts/accounts current-account-id] new-account)
+       ::save-account new-account})))
```

Comment:
Instead of this handler, which should just have public function taking db and settings (as only that is necessary) and producing re-frame effects data-structure:
```clj
(defn update-wallet-settings [{:account/keys [current-account-id accounts] :as db} settings]
  (let [new-account (-> (get accounts current-account-id)
                        (assoc :settings settings))]
    {:db            (assoc-in db [:accounts/accounts current-account-id] new-account)
     ::save-account new-account}))
```